packages/web3/src/transaction/utils.test.ts

```diff
@@ -121,11 +123,270 @@ describe('transaction utils', () => {
+            .mockResolvedValueOnce(mockConfirmed)
+        }
+      }
+      web3.setCurrentNodeProvider(mockProvider as any)
+
+      const result = await waitForTxConfirmation('test-tx-id', 1, 10)
+      expect(result.type).toBe('Confirmed')
+      expect(mockProvider.transactions.getTransactionsStatus).toHaveBeenCalledTimes(2)
+    })
+
+    it('should handle transaction errors', async () => {
+      const mockProvider = {
+        transactions: {
+          getTransactionsStatus: jest.fn().mockRejectedValue(new Error('Transaction not found'))
+        }
+      }
+      web3.setCurrentNodeProvider(mockProvider as any)
+
+      await expect(waitForTxConfirmation('invalid-tx', 1, 100)).rejects.toThrow('Transaction not found')
+    })
+
+    it('should handle multiple confirmation levels', async () => {
+      const mockSequence = [
+        { type: 'MemPooled' } as node.MemPooled,
+        {
+          type: 'Confirmed',
+          blockHash: '1234',
+          txIndex: 0,
+          chainConfirmations: 1,
+          fromGroupConfirmations: 1,
+          toGroupConfirmations: 1
+        } as node.Confirmed,
+        {
+          type: 'Confirmed',
+          blockHash: '1234',
+          txIndex: 0,
+          chainConfirmations: 3,
+          fromGroupConfirmations: 3,
+          toGroupConfirmations: 3
+        } as node.Confirmed
+      ]
+
+      const mockProvider = {
+        transactions: {
+          getTransactionsStatus: jest.fn()
+            .mockResolvedValueOnce(mockSequence[0])
+            .mockResolvedValueOnce(mockSequence[1])
+            .mockResolvedValueOnce(mockSequence[2])
+        }
+      }
+      web3.setCurrentNodeProvider(mockProvider as any)
+
+      const result = await waitForTxConfirmation('test-tx-id', 3, 10)
+      expect(result.type).toBe('Confirmed')
+      expect(result.chainConfirmations).toBe(3)
+      expect(mockProvider.transactions.getTransactionsStatus).toHaveBeenCalledTimes(3)
+    })
+
+    it('should handle network errors during polling', async () => {
+      const mockConfirmed: node.Confirmed = {
+        type: 'Confirmed',
+        blockHash: '1234',
+        txIndex: 0,
+        chainConfirmations: 1,
+        fromGroupConfirmations: 1,
+        toGroupConfirmations: 1
+      }
+  
+      const mockProvider = {
+        transactions: {
+          getTransactionsStatus: jest.fn()
+            .mockRejectedValueOnce(new Error('Network error'))
+            .mockResolvedValueOnce(mockConfirmed)
+            .mockResolvedValueOnce(mockConfirmed) 
+        }
+      }
+      web3.setCurrentNodeProvider(mockProvider as any)
+  
+      try {
+        await waitForTxConfirmation('test-tx-id', 1, 10)
+      } catch (error) {
+        const result = await waitForTxConfirmation('test-tx-id', 1, 10)
+        expect(result.type).toBe('Confirmed')
+        expect(mockProvider.transactions.getTransactionsStatus).toHaveBeenCalled()
+      }
+    })      
+
+  })
+
+  describe('transaction types', () => {
```

Comment:
You're right - the 'transaction types' tests aren't testing any actual logic since they're just verifying object fields. I will remove these tests and focus on testing the core functionality in utils.ts.